src/plugins/ExtendedProfileSampling.cpp

```diff
@@ -0,0 +1,862 @@
+/*
+ * ExtendedProfileSampling.cpp
+ *
+ *  Created on: Feb 2022
+ *      Author: homes
+ */
+
+#include "ExtendedProfileSampling.h"
+
+#include "particleContainer/ParticleContainer.h"
+#include "particleContainer/adapter/ParticlePairs2PotForceAdapter.h"
+#include "utils/Random.h"
+#include "Simulation.h"
+
+#include <math.h>
+
+
+ExtendedProfileSampling::ExtendedProfileSampling()
+    :
+    // Control: general
+    _binwidth(1.0f),
+    _startSampling(0ul),
+    _writeFrequency(10000ul),
+    _stopSampling(1000000000ul),
+    _singleComp(false),
+
+    // Higher moments
+    _sampleHigherMoms(false),
+
+    // Control: chemical potential
+    _sampleChemPot(false),
+    _lattice(true),
+    _factorNumTest(4.0f),
+    _samplefrequency(50ul)
+{}
+
+ExtendedProfileSampling::~ExtendedProfileSampling() {
+}
+
+void ExtendedProfileSampling::init(ParticleContainer* /* particleContainer */, DomainDecompBase* domainDecomp, Domain* domain) {
+
+    _globalBoxLength[0] = domain->getGlobalLength(0);
+    _globalBoxLength[1] = domain->getGlobalLength(1);
+    _globalBoxLength[2] = domain->getGlobalLength(2);
+
+    _numBinsGlobal = static_cast<uint16_t>(_globalBoxLength[1]/_binwidth);
+    if (_globalBoxLength[1]/_binwidth != static_cast<float>(_numBinsGlobal)) {
+        global_log->error() << "[ExtendedProfileSampling] Can not divide domain without remainder! Change binwidth" << std::endl;
+        Simulation::exit(-1);
+    }
+    _slabVolume = _globalBoxLength[0]*_globalBoxLength[2]*_binwidth;
+
+    if (_slabVolume < 1e-12) {
+        global_log->error() << "[ExtendedProfileSampling] Slab volume too small!" << std::endl;
+        Simulation::exit(-1);
+    }
+
+     // Entry per component and bin; 0 represents all components combined
+    _lenVector = (_singleComp) ? _numBinsGlobal : _numBinsGlobal * (domain->getNumberOfComponents()+1);
+
+    resizeVectors();
+    resetVectors();
+
+    // ??? TODO: CAN CELL PROCESSOR CHANGE DURING SIMULATION???
+    _cellProcessor = _simulation.getCellProcessor();
+    // ??? TODO: CAN PP HANDLER CHANGE DURING SIMULATION???
+    _particlePairsHandler = new ParticlePairs2PotForceAdapter(*domain);
+    // MolID is maximum possible number minus rank to prevent duplicate IDs
+    // Always insert molecule of first component
+    const unsigned long molID = std::numeric_limits<unsigned long>::max() - static_cast<unsigned long>(domainDecomp->getRank());
+    _mTest = Molecule(molID, &(_simulation.getEnsemble()->getComponents()->at(0)));
+}
+
+void ExtendedProfileSampling::readXML(XMLfileUnits& xmlconfig) {
+
+    xmlconfig.getNodeValue("binwidth", _binwidth);  // Default: 1.0
```

Comment:
not sure listing the defaults here is a good idea because it has to be kept consistent with the actual defaults which should be in the header at declaration.